social-api/src/main/java/org/geekbeacon/social/model/SocialActivity.java

```diff
@@ -0,0 +1,28 @@
+package org.geekbeacon.social.model;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.Builder;
+import lombok.Data;
+
+import java.net.URI;
+import java.util.Map;
+
+@Builder
+@Data
+@ApiModel("Model represents online social activity")
+public class SocialActivity {
+    @ApiModelProperty("originating URI for content")
+    private final URI sourceUri;
+    @ApiModelProperty("URL of media to display")
+    private final String mediaUrl;
+    @ApiModelProperty("text/comment/caption that goes with media if any")
+    private final String text;
+    @ApiModelProperty("dimensions of media")
+    private final Dimensions dimensions;
+    @ApiModelProperty("represents media dimensions")
+    private final boolean isVideo;
+    @ApiModelProperty("a hashmap of all socialLove data depending on the social media platform.  likes, comments, thumbs up, +1s etc")
+    private final Map<String, Object> socialLove;
```

Comment:
it's not stored in the DB.  It's a UI Model that is sent back to the user based on the data from the various API.  The intent was to make it flexible, so that if socialLove is called say a 'like' in one app, and another is a '+1s' or whatnot we can just add the item to the map without having an object that has:

likesCount, commentCounts, heartCounts, etc... based on social media source the content will change.

Reply:
Oh that makes sense, then.